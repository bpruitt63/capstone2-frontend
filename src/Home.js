import React from 'react';
import {Jumbotron} from 'reactstrap';
import {Link} from 'react-router-dom';
import './static/styles/Home.css';

function Home({username}) {

    return (
        <div className='container'>
            <Jumbotron>
                <h1>Welcome to Boatey!</h1>
                <p>The only place to plan your boat trips!
                    <span><br/>Before you get wet, get Boatey!</span>
                </p>
                {username &&
                    <p>Welcome back, {username}.
                        <span><br/>
                            Are you ready to{' '}
                            <Link to='/planner'>plan a new trip</Link>?
                        </span>
                    </p>    
                }
                {!username && 
                    <p>First time?{'  '}
                        <Link to='/register'>Create a Boatey account</Link>,
                        <span><br/>
                            or <Link to='planner'>start trip planning now!</Link>
                        </span></p>
                }
            </Jumbotron>
        </div>
    );
};

export default Home;