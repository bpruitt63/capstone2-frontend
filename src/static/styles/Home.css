.container {
    text-align: center;
}

.jumbotron {
    display: inline-block;
    width: 85%;
    background: rgba(255, 255, 255, 0.9);
    border-radius: 20px;
}

.jumbotron h1 {
    font-size: 6vw;
    margin: 20px 0px;
}

.jumbotron p a, .jumbotron p span a {
    text-decoration: none;
}