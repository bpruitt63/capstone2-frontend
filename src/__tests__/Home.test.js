import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Home from '../Home';
import { BrowserRouter } from 'react-router-dom';

test('renders', () => {
    render(<BrowserRouter><Home/></BrowserRouter>);
    expect(screen.getByText(/Create a Boatey account/)).toBeInTheDocument;
});

test('renders logged in', () => {
    render(<BrowserRouter><Home username={'test'}/></BrowserRouter>);
    expect(screen.getByText(/Welcome back, test/)).toBeInTheDocument;
});

test('matches snapshot', () => {
    const {asFragment} = render(<BrowserRouter><Home/></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
});