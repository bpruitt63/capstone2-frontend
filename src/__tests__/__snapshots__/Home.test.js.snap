// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="jumbotron"
    >
      <h1>
        Welcome to Boatey!
      </h1>
      <p>
        The only place to plan your boat trips!
        <span>
          <br />
          Before you get wet, get Boatey!
        </span>
      </p>
      <p>
        First time?  
        <a
          href="/register"
        >
          Create a Boatey account
        </a>
        ,
        <span>
          <br />
          or 
          <a
            href="/planner"
          >
            start trip planning now!
          </a>
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
